src/Agent.Worker/StepsRunner.cs

```diff
@@ -83,6 +83,10 @@ public async Task RunAsync(IExecutionContext jobContext, IList<IStep> steps)
                     HostContext.WritePerfCounter($"TaskStart_{taskStep.Task.Reference.Name}_{stepIndex}");
                 }
 
+                // Change the current job context to the step context.
+                var resourceDiagnosticManager = HostContext.GetService<IResourceMetricsManager>();
```

Comment:
Do I understand correctly that `resourceDiagnosticManager` will use step context each time? How is this context used? Could there be a cases when this switching is not needed?

Reply:
context is used for debug output. If we don't change context it writes log to JobInitialization context and makes UI redirect to that context page to show output to user.
Switching context helps to write resourcemetrics output on desired context log